### ISSUE TITLE
Unable to dismiss notification

### ISSUE THREAD

**Issue by tomme87** [folkehelseinstituttet/Fhi.Smittestopp.App#238]
I have a notification with "Smittestopp undersøker om du vær i nærheten brukere ..", but it never goes away, and I'm unable to dismiss it.

I usually get this notification periodically and it goes away, but now it stays forever. I have tried re-installing the app, but the problem re-appears as soon as I get the notification again.

For some reason the notification goes away if I toggle WiFi off/on, but re-appears again next time I get the notification (regardless if I'm on WiFi or not).

Android 10 (Oxygen OS 10.0.10.GM21BA) on OnePlus 7 Pro (GM1913).

**Comment by huaracheguarache** [comment]
I seem to be experiencing the same issue on my Samsung Galaxy S20 FE 4G with Android 11. The suggested workaround of toggling Wi-Fi off and on again made the notification disappear. I've only experienced this issue once, so I'm unable to confirm whether any subsequent notifications are undismissable.

**Comment by nc-kano** [comment]
Thank you for your messages. We have also observed this issue by ourselves. This is strange behavior as we had not introduced any update to the app since 21st of December. We are investigating the issue.

**Comment by nc-kano** [comment]
Apparently this issue happened also in German, Danish covid apps and in Google Messages app. It looks like the global issue introduced by Google. But we will continue investigating it to be sure the issue is not on our side.

**Comment by CUSL78** [comment]
@nc-kano We noticed it too it seems that it is on Android 11 Samsung 20Ultra, my husband experienced this after an Android update recently. I do not have this problem yet and I am still on Android 10 on my Samsung 20+.

**Comment by nc-kano** [comment]
@CUSL78 we are in contact with Google and we have shared the info about the issue so they are investigating it

**Comment by nc-kano** [comment]
Message from Google:
"Thank-you! Yes we're unfortunately seeing a few instances of seemingly similar behaviour across ENS apps.[...]" 
so they are on it.

**Comment by nc-kano** [comment]
@tomme87 @huaracheguarache @CUSL78 From another google mail: 
"Can you provide an adb bugreport? Perhaps take one now, and then also after a reboot (it seems likely to persist after a reboot and the bugreport will be "cleaner" after a reboot)"
Are you able to generate such report or you do not have tools and knowledge how to do that? Unfortunately we have not experienced this issue on our dev devices so we are unable to do it.

**Comment by nc-kano** [comment]
I have asked about help with reports in https://github.com/corona-warn-app/cwa-app-android/issues/2081

**Comment by nc-kano** [comment]
Here are the instructions how to generate report:
https://developer.android.com/studio/debug/bug-report

**Please do not share it in public. Make sure there are no personal data there and use a private message.**

**Comment by nc-kano** [comment]
From the German app:
"At the moment there is no exposure check possible with the Google ENF on android devices. We informed Google about this issue and they are working on a fix. We receive this issue also from multiple other countries.
When the problem is fixed from google side the app should recover themself with the next background check.
We will inform you when we receive updates."

**Comment by nc-kano** [comment]
"The Google Fix should be rolled out to all devices now.

Google Statement:
We are aware of an issue with the provideDiagnosisKeys() method of the Exposure Notifications API which affects all apps using the API. The issue became widespread around 12 Jan 22:00 UTC, and it causes the provideDiagnosisKeys() to appear to hang.
Tl;dr;
There is an issue currently affecting all users that prevent apps from matching the latest exposure data (since Jan 12th, 2021), but not from continuing broadcasting or scanning keys to devices in proximity
A fix is rolling out and will reach over 90% of your users by 1/14/2021 AM, EST time.
Users should not manually clear cache or storage, or uninstall the app. This won’t fix the issue, and it will incur exposure data loss on their devices

What is happening?

Calls to provideDiagnosisKeys() never return due to a defect originated on Jan 12th 2021, and in consequence matching does not occur. As a result, Exposure Notifications apps can’t match the latest exposure data (since Jan 12th only). This is the only EN API function affected, and we can confirm that:

Apps are handling exposure data in the device uninterrupted (broadcasting and scanning)
Uploading keys is working correctly and was never interrupted (reporting a case through the app)

We have determined the root cause, and a fix is already rolling out to users’ devices. Once the fix is installed in the device, calls to provideDiagnosisKeys() will have the expected behavior again. The rollout of the fix will reach over 90% of your users within 24 hrs."

**Comment by schox-nor** [comment]
Is there a way to check if handset is updated?

My Android phone got a "restart phone to finish update to Play Services" yesterday evening - but Exposure Notifications System (ENS) version is unchanged (18204714000) - And Google Play security update is 01.12.2020.

Notification still hangs - and log stating checks at
13 January 2021, 21:20
14 January 2021, 03:05

Notification active since 14 January 06:40

**Comment by hugocruz** [comment]
I had this issue recurrently, but since yesterday it stopped. ( Pixel 3 with latest android, all security updates and playstore updates ) 
These are the apps that might be related, not sure, given the description is not very obvious. 
![Screenshot_20210114-115606](https://user-images.githubusercontent.com/1942281/104582532-489c8b00-5660-11eb-83ca-e33a11e92cca.png)



Again, might be a coincidence, but I'll leave it here, might be useful.

**Comment by tomme87** [comment]
The issue seams to be gone for me now!

**Comment by nc-kano** [comment]
@schox-nor unfortunately we do not have more info from google. It seems they are rolling out the fix in different regions at different time so it may take some time to start working.

**Comment by schox-nor** [comment]
@nc-kano Have noticed the "notification behaviour" normalizing (including the log for "exposure checks")- so it might have been an isolated incident - or messaging que in google's api when all clients updating.
Issue is resolved for me ;)

**Comment by pavshr** [comment]
Since we are not receiving any more issues on support line and do not observe the issue on the test devices we have, as well as confirmation from Google that the bug is fixed, we are closing this issue. Feel free to reopen it or let us know if for some reason the issue is not resolved on your device.

Thank you everyone!
